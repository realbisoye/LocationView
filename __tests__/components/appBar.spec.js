import React from 'react'
import renderer from 'react-test-renderer'
import AppBar from '../../app/components/shared/appBar'

describe('GlobalAppBar', () => {
  const onMenuClick = jest.fn()
  const rendered = renderer.create(
    <AppBar
      title='Sample'
      menuIconName={'map-marker'}
      onMenuClick={onMenuClick}
    />
  )
  it('renders GlobalAppBar', () => {
    expect(rendered).toMatchSnapshot()
  })

  it('call action on menu click', () => {
    const menuButton = rendered.root.findByProps({testID: 'menuButton'})
    menuButton.props.onPress()
    expect(onMenuClick).toHaveBeenCalled()
  })
})
