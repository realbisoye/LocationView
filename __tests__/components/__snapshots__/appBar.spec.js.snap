// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalAppBar renders GlobalAppBar 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "maxHeight": "5%",
        "paddingHorizontal": 20,
        "top": 0,
        "width": "100%",
      },
      Object {
        "backgroundColor": "#88BF41",
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="menuButton"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#171914",
            "fontSize": 25,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      Ôçé
    </Text>
  </View>
  <Text
    style={
      Object {
        "fontSize": 22,
        "fontWeight": "bold",
        "marginLeft": "22%",
        "zIndex": 1000,
      }
    }
    testID="apptitle"
  >
    Sample
  </Text>
</View>
`;
